### PR TITLE
create uni-v3 gtc position

### PR DESCRIPTION
**Pull request by jalbrekt85**
handles #1122, includes a `get_amount_out` method in the uni-v3 class to help dynamically set ranges

### CODE REVIEW

File: scripts/issue/1122/create_gtc_position.py [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+
+    range_0 = gtc_rate * LOWER_RANGE_MULTIPLIER / 1e18
+    range_1 = gtc_rate * UPPER_RANGE_MULTIPLIER / 1e18
+
+    vault.uni_v3.mint_position(
+        r.uniswap.v3pool_eth_gtc,
+        range_0,
+        range_1,
+        weth.balanceOf(vault),
+        gtc.balanceOf(vault),
```
Comment (gosuto-inzasheru) [inline scripts/issue/1122/create_gtc_position.py:37]: this will need to be done dusty, since outcome of swap is not 100% deterministic in our case

**Review by gosuto-inzasheru — CHANGES_REQUESTED** [review]
code looks clean! just be wary of diff between posting and execing :)

File: scripts/issue/1122/create_gtc_position.py [hunk 2]
```diff
@@ -0,0 +1,41 @@
+from great_ape_safe import GreatApeSafe
+from helpers.addresses import r
+
+LOWER_RANGE_MULTIPLIER = 0.9
+UPPER_RANGE_MULTIPLIER = 2
+GTC_SELL_PCT = 0.16
```
Comment (petrovska-petro) [inline scripts/issue/1122/create_gtc_position.py:6]: this i think should be 14.5% instead
```suggestion
GTC_SELL_PCT = 0.145
```

Comment (jalbrekt85) [reply scripts/issue/1122/create_gtc_position.py]: updated in fdcaa9d

File: scripts/issue/1122/create_gtc_position.py (hunk 1 above)
Comment (jalbrekt85) [reply scripts/issue/1122/create_gtc_position.py]: good catch, using `0.97`: 517e2e9

Comment (petrovska-petro) [reply scripts/issue/1122/create_gtc_position.py]: should not be only dusty for weth?

Comment (jalbrekt85) [reply scripts/issue/1122/create_gtc_position.py]: yup, dont see why its needed for gtc

Comment (jalbrekt85) [reply scripts/issue/1122/create_gtc_position.py]: > should not be only dusty for weth?

e5833d1

**Review by petrovska-petro — APPROVED** [review]

**Pull request merged**